stdlib/public/core/ManagedBuffer.swift

```diff
@@ -136,6 +136,29 @@ open class ManagedBuffer<Header, Element> {
   }
 }
 
+public extension ManagedBuffer {
+  @inline(never)
+  final class func reallocate<Buffer: ManagedBuffer>(
+    buffer: inout Buffer,
+    count: Int,
+    newMinimumCapacity: Int
+    ) {
+      precondition(_isBitwiseTakable(Header.self))
+      precondition(_isBitwiseTakable(Element.self))
+      precondition(isKnownUniquelyReferenced(&buffer))
+      precondition(!_hasSideTable(&buffer))
+
+      let newSizeInBytes = MemoryLayout<Header>.stride
+        + newMinimumCapacity * MemoryLayout<Element>.stride
+
+      withUnsafePointer(to: &buffer) { bufferPtr in
```

Comment:
@mikeash 🙈. Is this really the best way of doing this? I think this is pretty wrong as it doesn't guarantee to give me the right pointer but it does work...

Reply:
@mikeash checked some more but couldn't find anything that would allow me to rewrite the pointer of `inout Buffer` to a new value without messing the with ref counts (apart from the unsafeBitCast obvs)...